libs/community/tests/integration_tests/chat_models/test_deepinfra.py

```diff
@@ -63,3 +75,51 @@ async def test_async_chat_deepinfra_streaming() -> None:
     assert isinstance(generation, ChatGeneration)
     assert isinstance(generation.text, str)
     assert generation.text == generation.message.content
+
+
+def test_chat_deepinfra_bind_tools() -> None:
+    class Foo(BaseModel):
+        pass
+
+    chat = ChatDeepInfra(
+        max_tokens=10,
+    )
+    tools = [Foo]
+    chat_with_tools = chat.bind_tools(tools)
+    assert isinstance(chat_with_tools, RunnableBinding)
+    chat_tools = chat_with_tools.tools
+    assert chat_tools
+    assert chat_tools == {
+        "tools": [
+            {
+                "function": {
+                    "description": "",
+                    "name": "Foo",
+                    "parameters": {"properties": {}, "type": "object"},
+                },
+                "type": "function",
+            }
+        ]
+    }
+
+
+def test_tool_use() -> None:
+    llm = ChatDeepInfra(model="meta-llama/Meta-Llama-3-70B-Instruct", temperature=0)
+    llm_with_tool = llm.bind_tools(tools=[GenerateMovieName], tool_choice=True)
+    msgs: List = [
+        HumanMessage(content="It should be a movie explaining humanity in 2133.")
+    ]
+    ai_msg = llm_with_tool.invoke(msgs)
+
+    assert isinstance(ai_msg, AIMessage)
+    assert isinstance(ai_msg.tool_calls, list)
+    assert len(ai_msg.tool_calls) == 1
+    tool_call = ai_msg.tool_calls[0]
+    assert "args" in tool_call
+
+    tool_msg = ToolMessage(
+        content="Year 2133",
+        tool_call_id=ai_msg.additional_kwargs["tool_calls"][0]["id"],
+    )
+    msgs.extend([ai_msg, tool_msg])
```

Comment:
@ovuruska do we also need to update `_convert_message_to_dict` to read `.tool_calls` off of incoming AIMessages and structure them into API requests?
https://github.com/langchain-ai/langchain/blob/b7ad23af53dc52e35ff800ba5ec1bab964716583/libs/community/langchain_community/chat_models/deepinfra.py#L129-L132

Reply:
@ccurme Fixed the issue. Thanks for noticing it.